repo_commits.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2014 The Gogs Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package gogs
+
+import (
+	"fmt"
+	"time"
+)
+
+type Sha1 struct {
+	Sha1  string             	  `json:"sha1"`
+}
+
+type Signature struct {
+	Email   string            `json:"email"`
+	Name    string            `json:"name"`
+	When 	time.Time         `json:"when"`
+}
+
+type Commit struct {
+	ID      		string             	   `json:"id"`
+	Author			Signature			   `json:"author"`
+	Committer		Signature			   `json:"committer"`
+	CommitMessage	string			   	   `json:"commit_message"`
+}
+
+type DiffLine struct {
+	LeftIdx  int		`json:"left_idx"`
+	RightIdx int		`json:"right_idx"`
+	Type     string		`json:"type"`
+	Content  string		`json:"content"`
+}
+
+type DiffSection struct {
+	Name  string		`json:"name"`
+	Lines []*DiffLine	`json:"lines"`
+}
+
+type DiffFile struct {
+	Name               string				`json:"name"`
+	Index              int					`json:"index"`
+	Addition		   int 					`json:"addition"`
+	Deletion 		   int					`json:"deletion"`
+	Type               string				`json:"type"`
+	IsCreated          bool					`json:"created"`
+	IsDeleted          bool					`json:"deleted"`
+	IsBin              bool					`json:"bin"`
+	Sections           []*DiffSection		`json:"sections"`
+}
+
+type Diff struct {
```

Comment:
No. It is used by Gogs, and is needed for my gogs pull request. See the DiffRange function in the that pull request. 

This is also why the gogs pull request has dependency to kiliit/go-gogs-client. If it hasn't then compile fails.